### ISSUE TITLE
Serial Plotter (IDE 2.0) issue

### ISSUE THREAD

**Issue by kubichekster** [arduino/arduino-ide#803]
**Is your feature request related to a problem? Please describe.**

The new Serial Plotter displays only 50 data values, compared to the previous version, which displayed 500 values.
This is a problem because events stream by 10 times faster and are much harder to see. With 500 points we see a much clearer time history of the physical measurements we're making. I use the Plotter a lot in my K-12 STEM workshops, and this will really have a negative impact.

**Describe the solution you'd like**
Change the number of plot values back to 500. Or make it a user selectable option

**Describe alternatives you've considered**
I can modify my old programs to slow Serial.print outputs so the data stream doesn't disappear so quickly. It kinda works but it's suboptimal.
 
**Additional context**

**Comment by ratio-x** [comment]
urgently expected...

**Comment by IanBUK** [comment]
Export to Excel would be nice as well.

**Comment by fralbo** [comment]
In fact, the best would be to let us define the horizontal scale.

**Comment by InstantMuffin** [comment]
I, too, stumbled upon this.

Obviously it would be nice to have specific features like mentioned above, but I think just to have on par featureset with the former IDE it would be wise to readd display of 500 data points like in the earlier versions. It would be quite odd to have a successor that does not have or has less of a specific feature than the predecessor.

I tried changing this line
https://github.com/arduino/arduino-serial-plotter-webapp/blob/04a460e0c21609a63c4064f561990c0128c5a0ed/src/ChartPlotter.tsx#L46
in the bundled package that is being fetched/created when building arduino-ide. The build process did not edit the file back to what it was before, however, unfortunately, it did not increase the number of data points.
Being unfamiliar with the build process, I don't know if this is in fact the wrong line of code to adapt, or if the change just doesn't make it into the final arduino-ide build for some reason.

The change in itself would seem to be rather trivial, given that and the featureset of the predecessor I would really like to see the this implemented before the final release.

**Comment by SaturdayScience** [comment]
I'm running rc9.4 on Linux Mint 21 and the plotter frustrates me. With data once per second plotter shows only 5 X-axis data sets no matter how large I make the window. At fast rate such as 100 data sets / second it shows 50. I can't seem to display 500 seat.

**Comment by Farhankhosravi** [comment]
Hi..
yup, this is so URGENT and needed to make plotter scalable... 50 is really small!
only make two sliders to change x/y zoom manually! not a hard work, but VERY life savior.

I'm waiting for this too, hope they fix this ASAP.

**Comment by SvenRosvall** [comment]
Another vote for this enhancement.

**Comment by fralbo** [comment]
Anyone in the development team to take care about our requests

**Comment by rhettherman** [comment]
Either going back to the 500-point serial plotter, or (better yet) allowing for perhaps several options for the number of points to plot, would be most appreciated.

**Comment by rhettherman** [comment]
Either going back to the 500-point serial plotter, or (better yet) allowing for perhaps several options for the number of points to plot, would be most appreciated.

**Comment by pointsplitter** [comment]
Either going back to the 500-point serial plotter, or (better yet) allowing for perhaps several options for the number of points to plot, would be most appreciated.

**Comment by ArnieO** [comment]
Yes, please do the requested enhancement / fix!
I fortunately have IDE 1.8.19 installed in parallel - it still displays 500 points.

**Comment by bruno-zarokian** [comment]
To add to that, my plotter would sometimes display only 5 samples! Some random series of action usually gets it back to 50 samples. Seems like jumping between multiple projects does/undoes the issue but haven't explored well enough. All my projects use serial over USB btw.

**Comment by per1234** [comment]
>my plotter would sometimes display only 5 samples! Some random series of action usually gets it back to 50 samples. Seems like jumping between multiple projects does/undoes the issue but haven't explored well enough.

Thanks for your report @bruno-zarokian. If you are able to find a procedure that can be followed to reproduce this bug, please submit a dedicated issue for it.

**Comment by InstantMuffin** [comment]
I expected a few comments in regards to parity to the 1.X feature state, but I didn't expect so many.

I, too, still want this to be an par or to allow for more samples than 500.

**Comment by per1234** [comment]
Hi all. We really appreciate the feedback about how the Arduino IDE 2.x Serial Plotter can be improved. However, once the message has been received, additional comments that don't add any new information can add up to a lot of notifications for those of us who watch this repository to keep up with the development work.

If you only want to express support then I suggest the use of [GitHub's "Reactions" feature](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/).

Discussion is always welcome over on the Arduino Forum:

https://forum.arduino.cc/

Thanks in advance for your cooperation.
Per

**Comment by Alex-Tsyganok** [comment]
The serial plotter is a web application that is installed as part of Arduino IDE since it's running as a web application, bundled source code can be modified on a local machine.

The package can be found in the installation folder of Arduino IDE 
`\Arduino IDE\resources\app\node_modules\arduino-serial-plotter-webapp\build\static\js\`
`main.35ae02cb.chunk.js` can be modified to change values threshold value, need to replace `U=Object(o.useState)(50)` with `U=Object(o.useState)(5000)`  it will increase maximum data points to 5000. 

Tested with Arduino IDE 2.0.2 version.
![image](https://user-images.githubusercontent.com/5132633/205741021-223c3130-0a71-4264-9ac1-38eb364deb82.png)

**Comment by fralbo** [comment]
🤪
Thanks Alex.
Better layer than never...

**Comment by kubichekster** [comment]
Thanks Alex - nice solution! I will give it a try.
However, I hope the developers will build this into the IDE asap. I work with K-12 teachers across the state and the fix needs to be plug-and-play before they can use it at their schools. They also need a portable version, but that's another story.
Thanks again

________________________________

**Comment by falk-stefan** [comment]
@Alex-Tsyganok Thank you for that quick fix!

But I have to tell you that I felt the strong urge to wash my hands after that 😂

**Comment by zapta** [comment]
I tried to modified the js file but not luck here, still only 50 points.  Windows 10, Arduino IDE 2.0.3, notepad++ editor.  I also restarted Windows, in case there is some caching involved. Any suggestions?

![image](https://user-images.githubusercontent.com/2184769/212458410-1b6814ea-e030-4b2d-9685-c07f096a2a17.png)

**Comment by Leomirdroid** [comment]
Worked for me as well. To save the changes, I had to open notepad in admin mode, change the value and save (without the txt ending by clicking on all files next to the save button). @zapta did it work for you now? Did you open the correct Arduino version? Did you try uninstalling all older versions and make a clear reinstall?

**Comment by Ralf-W** [comment]
@Alex-Tsyganok Thank you a lot!

**Comment by RolfHut** [comment]
I'll briefly add that while the fix by @Alex-Tsyganok works on my Mac, this is not a solution for teachers (like me) who rely on the IDE in their teaching. Students can't be expected to dig this deep into the code of an application. I have to resort to either advising students to fall back on the old IDE, or use third party things (like processing), but neither of this is good teaching.

Can the fix by @Alex-Tsyganok just be implemented for now, while people work on a more lasting solution like allowing users to change the x-axis?

**Comment by Dlloydev** [comment]
As it stands now, the plot width auto scales to full width as the data fills to 50 samples, then the plotter becomes a sliding window.

![Plotter](https://user-images.githubusercontent.com/63488701/236367038-71621433-2d42-4e88-9578-b58444e7ac42.gif)

It would be useful to have the plot width auto scale to full width as the data fills to 500 samples (or a user set limit), then the sliding window activates.

**Comment by Farhankhosravi** [comment]
Hello again...
Came to ask that how long this feature will be take?! Just asking for more samples like IDE 1.8 and older! It's really necessary but after a long while, it doesn't added to IDE V2...

Still waiting for such small feature. Please take a look!

**Comment by zapta** [comment]
@per1234, @AlbyIanna, any news on this issue?  It was assigned a year ago but so far there is no indication here of any progress.  Can you please look into it?

**Comment by Pkanna10** [comment]
> The serial plotter is a web application that is installed as part of Arduino IDE since it's running as a web application, bundled source code can be modified on a local machine.
> 
> The package can be found in the installation folder of Arduino IDE `\Arduino IDE\resources\app\node_modules\arduino-serial-plotter-webapp\build\static\js\` `main.35ae02cb.chunk.js` can be modified to change values threshold value, need to replace `U=Object(o.useState)(50)` with `U=Object(o.useState)(5000)` it will increase maximum data points to 5000.
> 
> Tested with Arduino IDE 2.0.2 version. ![image](https://user-images.githubusercontent.com/5132633/205741021-223c3130-0a71-4264-9ac1-38eb364deb82.png)

I just tested, and the solution works!! Thanks a lot. Any idea on how to make it go slower?

**Comment by kubichekster** [comment]
OK, thanks, that sounds useful!
But I have a newbie question: What do I do after I make the change to the .js file? Do I need to recompile or rebuild something in order to update the IDE? Any details would be appreciated.
As a general comment, since the fix seems fairly trivial it seems crazy they don't default the number of data points to the version 1.x value (i.e., 500) so all the legacy Arduino programs would still work as designed.
thanks again



________________________________

**Comment by per1234** [comment]
Hi @kubichekster. This issue is used to track the request for the Arduino developers to increase the number of displayed data points or to make it user configurable. Comments that add unique information that is relevant to that development work are welcome, but this is not an appropriate place to get assistance with [the hack that was shared](https://github.com/arduino/arduino-ide/issues/803#issuecomment-1338149431).

If anyone needs assistance or has questions about that hack, please post on Arduino Forum:

https://forum.arduino.cc/

I'm sure we'll be able to help you out over there.

**Comment by samstafford** [comment]
@Alex-Tsyganok  thank you for this patch; was very helpful to me.

**Comment by per1234** [comment]
A PR for adding a menu to control the number of data points has been submitted to the repository where the Serial Plotter codebase is hosted:

https://github.com/arduino/arduino-serial-plotter-webapp/pull/32

Please refrain from commenting on that PR thread unless you have some significant and unique information to add. If you only want to express your support for the PR or a comment in the thread, use the ["Reactions" feature](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to add a 👍, etc. If the thread becomes a source of notification noise due to the addition of comments that don't add significant and unique information, as happened here, I will be forced to lock the conversation, which might prevent the addition of comments that are actually of value.
